src/precice/bindings/python_future/precice_future.pyx

```diff
@@ -98,134 +98,140 @@ cdef class Interface:
 
     # data access
     # hasData
-    def has_data (self, str data_name, int mesh_id):
+    def has_data (self, str data_name, mesh_id):
         return self.thisptr.hasData(convert(data_name), mesh_id)
 
-    def get_data_id (self, str data_name, int mesh_id):
+    def get_data_id (self, str data_name, mesh_id):
         return self.thisptr.getDataID (convert(data_name), mesh_id)
 
-    def map_read_data_to (self, int to_mesh_id):
+    def map_read_data_to (self, to_mesh_id):
         self.thisptr.mapReadDataTo (to_mesh_id)
 
-    def map_write_data_from (self, int from_mesh_id):
+    def map_write_data_from (self, from_mesh_id):
         self.thisptr.mapWriteDataFrom (from_mesh_id)
 
-    def write_block_vector_data (self, int data_id, value_indices, values):
+    def write_block_vector_data (self, data_id, value_indices, values):
+        size, dimensions = values.shape
+        assert(dimensions == self.get_dimensions())
         cdef np.ndarray[int, ndim=1] _value_indices = np.ascontiguousarray(value_indices, dtype=np.int32)
-        cdef np.ndarray[double, ndim=1] _values = np.ascontiguousarray(values, dtype=np.double)
-        assert(_values.size / self.get_dimensions() == _value_indices.size)
+        cdef np.ndarray[double, ndim=1] _values = np.ascontiguousarray(values.flatten(), dtype=np.double)
+        assert(size == _value_indices.size)
         size = value_indices.size
         self.thisptr.writeBlockVectorData (data_id, size, &_value_indices[0], &_values[0])
 
-    def write_vector_data (self, int data_id, value_index, value):
+    def write_vector_data (self, data_id, value_index, value):
+        dimensions = value.size
+        assert(dimensions == self.get_dimensions())
         cdef np.ndarray[np.double_t, ndim=1] _value = np.ascontiguousarray(value, dtype=np.double)
         self.thisptr.writeVectorData (data_id, value_index, &_value[0])
 
-    def write_block_scalar_data (self, int data_id, value_indices, values):
+    def write_block_scalar_data (self, data_id, value_indices, values):
         cdef np.ndarray[int, ndim=1] _value_indices = np.ascontiguousarray(value_indices, dtype=np.int32)
         cdef np.ndarray[double, ndim=1] _values = np.ascontiguousarray(values, dtype=np.double)
         assert(_values.size == _value_indices.size)
         size = value_indices.size
         self.thisptr.writeBlockScalarData (data_id, size, &_value_indices[0], &_values[0])
 
-    def write_scalar_data (self, int data_id, int value_index, double value):
+    def write_scalar_data (self, data_id, value_index, double value):
         self.thisptr.writeScalarData (data_id, value_index, value)
 
-    def read_block_vector_data (self, int data_id, value_indices):
+    def read_block_vector_data (self, data_id, value_indices):
         cdef np.ndarray[int, ndim=1] _value_indices = np.ascontiguousarray(value_indices, dtype=np.int32)
         size = _value_indices.size
```

Comment:
Should be able to handle tuples and lists

Reply:
I do not think that there is any problem in this function.